test/soak/serve_hostnames/serve_hostnames.go

```diff
@@ -112,7 +112,16 @@ func main() {
 	// Make a unique namespace for this test.
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
 	ns := "serve-hostnames-" + strconv.Itoa(r.Int()%10000)
-	glog.Infof("Using namespace %s for this test.", ns)
+	glog.Infof("Creating namespace %s", ns)
+	// Create the namespace
+	if _, err = c.Namespaces().Create(&api.Namespace{ObjectMeta: api.ObjectMeta{Name: ns}}); err != nil {
```

Comment:
Nit: I'd prefer to have the server generate the namespace name. This guarantees no collisions.

```
got, err := c.Namespaces().Create(&api.Namespace{ObjectMeta: api.ObjectMeta{GenerateName: "serve-hostnames-"}})
if err ... { ... }
ns := got.Name
```